### ISSUE TITLE
You can not use codeformers face restoration if gfpgan installed

### ISSUE THREAD

**Issue by Vargol** [invoke-ai/InvokeAI#768]
**Describe your environment**
- GPU:  mps
- VRAM: n/A but 8Gb M1
- CPU arch: arm
- OS: macOS
- Python: pip installed
- Branch: development
- Commit: commit cd8be1d0e9f6fccad5f796668e8d2e0611bbdb0d (HEAD -> development, origin/development)

**Describe the bug**
If both the codeformers model and gfpgan are installed then the code will alway use gfpgan.

Code has a logic 'bug' that is if self.gfpgan is defined it will not use codeformers
```
                    if self.gfpgan is not None or self.codeformer is not None:
                        if self.gfpgan is None:
                            if facetool == 'codeformer':
                                if self.codeformer is not None:

```

If both are installed both self.gfpgan and self.codeformer are not `None` as self.gfpgan is not `None` is does not use codeformer
face restoration.

**To Reproduce**
Check you have GFPGAN installed and working
```
"a young helen mirren in chain mail armour" -s 50 -S 64196915 -W 256 -H 256 -C 15.5 -A k_lms  -G 0.7
```

You will get 
```
>> GFPGAN - Restoring Faces for image seed:64196915
```
In the output.

Install the codeformers restoration  model 
and attempt the same prompt using codeformer as the face tool
```
"a young helen mirren in chain mail armour" -s 50 -S 64196915 -W 256 -H 256 -C 15.5 -A k_lms -ft codeformer -cf 0.7 -G 0.7
```

You will still get 
```
>> GFPGAN - Restoring Faces for image seed:64196915
```
In the output.

**Expected behavior**
```
"a young helen mirren in chain mail armour" -s 50 -S 64196915 -W 256 -H 256 -C 15.5 -A k_lms -ft codeformer -cf 0.7 -G 0.7
```

wil get 
```
>> CodeFormer - Restoring Faces for image seed:64196915
```
In the output.

**Comment by timdesrochers** [comment]
This has been working for me... I have the GFPGAN repo cloned separately and run things through to compare; outputs from -ft codeformer appear to be distinct, and similar to RestoreFormer, related to CF, which GFPGAN can use as a model. 

Is it just the case that CodeFormer is a model and GFPGAN is still the thing using it?

**Comment by Vargol** [comment]
Don't know.


I've stuck a  print in the code above and I alway get a value for self.gfpgan

```
<ldm.dream.restoration.gfpgan.GFPGAN object at 0x11c1cee00>
<ldm.dream.restoration.codeformer.CodeFormerRestoration object at 0x28142fb20>

```

So it never goes in the codeformer call

I've had to comment out the is None: bit to test  codeformer (as it has a issue on MPS, well it looks like its actually facexlib that does, waiting for a decent face to restore to test while typing this)

**Comment by Vargol** [comment]
looks like a commit around 11 hours ago is the cause.

https://github.com/invoke-ai/InvokeAI/commit/cd8be1d0e9f6fccad5f796668e8d2e0611bbdb0d

totally changed how that logic worked.
Code in  ldm/generate.py

**Comment by timdesrochers** [comment]
> This has been working for me... I have the GFPGAN repo cloned separately and run things through to compare; outputs from -ft codeformer appear to be distinct, and similar to RestoreFormer, related to CF, which GFPGAN can use as a model.
> 
> Is it just the case that CodeFormer is a model and GFPGAN is still the thing using it?

I was wrong. In fact, the -cf and -ft switches aren't even winding up in my dream_log:

CodeFormer via -ft:
`[...] photograph by Leibovitz, sunny, huge dark green leaves, 8K HDR" -s 8 -S 444 -W 512 -H 512 -C 10.0 -A k_euler -V 744951550:0.33`
![cf-not](https://user-images.githubusercontent.com/8886024/192049656-16a1d2c9-a0a8-4d62-b192-1297183b060b.jpg)

GFPGAN via -G:
`[...] photograph by Leibovitz, sunny, huge dark green leaves, 8K HDR" -s 8 -S 444 -W 512 -H 512 -C 10.0 -A k_euler -G 1.0 -V 744951550:0.33`
![gfp](https://user-images.githubusercontent.com/8886024/192049658-f07b07c5-7c17-4745-9a16-c6840a97678c.jpg)

**Comment by blessedcoolant** [comment]
Thanks for pointing it out. Should be fixed in #776. Please pull that and check and let me know if there's any other use cases that I might have missed.
